src/app/components/Sidebar.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+
+import React from "react";
+import {
+  useSidebarStateContext,
+  useSidebarActionsContext,
+} from "@context/SidebarContext";
+import {
+  Drawer,
+  Box,
+  Typography,
+  List,
+  ListItemButton,
+  Divider,
+} from "@mui/material";
+import theme from "theme";
+import { getCourses } from "@services/client/course";
+import { useUserSession } from "@context/UserSessionContext";
+import { useRouter } from "next/navigation";
+import { usePathname } from "next/navigation";
+import { IdentifiableCourse } from "@interfaces/type";
+import MenuButton from "./buttons/MenuButton";
+import ModeEditOutlinedIcon from "@mui/icons-material/ModeEditOutlined";
+
+const Sidebar = React.memo(() => {
+  const [courses, setCourses] = React.useState<IdentifiableCourse[]>([]);
+  const { isOpen } = useSidebarStateContext();
+  const { closeSidebar } = useSidebarActionsContext();
+  const { user } = useUserSession();
+  const { courses: courseIds = [] } = user || {};
+  const router = useRouter();
+  const pathname = usePathname();
+
+  React.useEffect(() => {
+    const loadCourses = async () => {
+      const allCourses = await getCourses();
+      const filteredCourses = allCourses
+        .filter((course) => courseIds.includes(course.id))
+        .map((course) => course);
+      setCourses(filteredCourses);
+    };
+    loadCourses();
+  }, [user]);
+
+  const handleCourseClick = (course: IdentifiableCourse) => {
+    const isCourseRoute = /\/[a-zA-Z]+\d+/.test(pathname);
+
+    const basePath = isCourseRoute
+      ? pathname.replace(/\/[a-zA-Z]+\d+.*$/, "")
+      : pathname;
+
+    router.push(`${basePath}/${course.id}`);
+    closeSidebar();
+  };
+
+  const handleManageCoursesClick = () => {
+    const isCourseRoute = /\/[a-zA-Z]+\d+/.test(pathname);
+
+    const basePath = isCourseRoute
+      ? pathname.match(/(.*\/[a-zA-Z]+\d+)/)?.[0]
+      : `${pathname}/${courses[0].id}`;
+
+    router.push(`${basePath}/profile/edit`);
+    closeSidebar();
+  };
+
+  const formattedCourseName = (course: IdentifiableCourse) => {
+    const formattedCourseId = course.id
+      .toUpperCase()
+      .replace(/([a-zA-Z]+)(\d+)/, "$1 $2");
```

Comment:
Double checking - should there be a space?

Reply:
Yeah I think so, according to Figma at least:

<img width="450" alt="image" src="https://github.com/user-attachments/assets/713f38e2-44ee-4e5d-ad98-3cdf6187f574" />

Reply:
I see some wireframe has it together - might be worth confirming with Fa and Sarah
<img width="304" alt="image" src="https://github.com/user-attachments/assets/ff616ab4-ebb1-4cb0-8cd9-c77f0d810caa" />